### ISSUE TITLE
Type cast error when running addDemographics() and addAge() - issue with translation to Redshift SQL?

### ISSUE THREAD

**Issue by pa-nathaniel** [darwin-eu-dev/PatientProfiles#341]
Hi PatientProfiles team,

Really excited to use the package! Unfortunately I ran into an error when trying `addDemographics()` and `addAge()` I was hoping to get some guidance on.

I have an object called `x` which is a list containing `tables_cdm`, a `cdm_reference` object created from `CDMConnector::cdm_from_con()`, and `cohort_tbl`, a connection to a a cohort table created by `DBI::dbConnect()`. 

The tables are on Redshift (Update: Now I think this is the culprit!)

Unfortunately, when I call `addDemographics()`, I get the following error about a mismatch in types (I also get the same error when running `addAge()`)

```
addDemographics(x$cohort_tbl, cdm = x$tables_cdm)
Error in `db_save_query.DBIConnection()`:
! Can't save query to "dbplyr_011".
Caused by error:
! Failed to fetch row: ERROR:  function concat_ws("unknown", character varying, "unknown", character varying, "unknown", character varying) does not exist
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
Run `rlang::last_trace()` to see where the error occurred.
```

Any tips on what could be causing this issue and how to debug?

**Comment by pa-nathaniel** [comment]
Did some digging through the underlying SQL query and I suspect this line is the source of the issue:

```
    CAST(CONCAT_WS('', "year_of_birth1", '-', "month_of_birth1", '-', "day_of_birth1") AS DATE) AS "date_of_birth"
```

**Comment by catalamarti** [comment]
Hi @pa-nathaniel happy to see that people use our packages. So at the moment we had some data partners using Redshift and they didn't report anything like that.

First to confirm, can I confirm that you connect to the redshift following something similar to this instructions? https://darwin-eu.github.io/CDMConnector/articles/a04_DBI_connection_examples.html#redshift

How do you create your cohort? if it is already instantiated in the database you can read it at the cdm object creation:
```
cdm <- cdm_from_con(con, cdm_schema = "main", write_schema = "mian", cohort_tables = "cohort_name")
```

Otherwise if you need to instantiate in from a json file you can use `generateCohortSet`:
```
cohortSet <- readCohortSet(system.file("cohorts2", package = "CDMConnector"))
cdm <- generateCohortSet(cdm, cohortSet, name = "cohort_name")
```

So basically can you try to run something like this?
```
library(DBI)
library(CDMConnector)
library(PatientProfiles)

con <- DBI::dbConnect(RPostgres::Redshift(),
                      dbname   = "....",
                      host     = "...",
                      port     = "...",
                      user     = "...",
                      password = "...")

cdm <- cdm_from_con(con, cdm_schema = "...", write_schema = "...",  cohort_tables = "cohort_name")

cdm$cohort_name %>%
  addDemographics()
```

Could you also please install the last available versions? CDMConnector 1.1.2, PatientProfiles 0.3.1
